dictionary.cc

```diff
@@ -171,6 +171,11 @@ vector< wstring > Class::getAlternateWritings( wstring const & )
   return vector< wstring >();
 }
 
+QString Class::getContainingFolder()
+{
+  return QFileInfo( QString::fromStdString( dictionaryFiles[ 0 ] ) ).absolutePath();
+}
```

Comment:
if dictionaryFiles[0] is `c:\abc\a.txt`
the original code should return c:\abc

I'm not sure the above code can return c:\abc ,shouldn't return c:\abc\a.txt

Reply:
or I made a mistake. Counter-intuitive

Reply:
at `qtbase/src/corelib/io/qfileinfo.cpp`

```
QString QFileInfo::absolutePath() const
{
    Q_D(const QFileInfo);

    if (d->isDefaultConstructed)
        return ""_L1;
    return d->getFileName(QAbstractFileEngine::AbsolutePathName);
}
```
The getFilename is 

```
QString QFileInfoPrivate::getFileName(QAbstractFileEngine::FileName name) const
{
        ...
        switch (name) {
            case QAbstractFileEngine::AbsolutePathName: {
                QFileSystemEntry entry = QFileSystemEngine::absoluteName(fileEntry);
                if (cache_enabled) { // be smart and store both
                    fileNames[QAbstractFileEngine::AbsoluteName] = entry.filePath();
                    fileNames[QAbstractFileEngine::AbsolutePathName] = entry.path();
                }
                if (name == QAbstractFileEngine::AbsoluteName)
                    ret = entry.filePath();
                else
                    ret = entry.path();
                break;
            }
            default: break;
        }
        ...
}
```

Which leads to here that chop the file name and only have the path part

```
QString QFileSystemEntry::path() const
{
    findLastSeparator();
    if (m_lastSeparator == -1) {
        return QString(u'.');
    }
    if (m_lastSeparator == 0)
        return QString(u'/');
        
    return m_filePath.left(m_lastSeparator);
}
```

Reply:
> Counter-intuitive

Yes, it is.

There are two methods:

* QFileInfo::absoluteFilePath()
* QFileInfo::absolutePath()

The 1st one contains the file name.

Reply:
just got this ,
![image](https://user-images.githubusercontent.com/105986/231944844-331b1468-1ba5-4c32-90a5-9765f46f9173.png)